internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
+		}
+
+		if d.HasChange("quota") {
+			log.Printf("[DEBUG] Updating the Quota for %s", id)
+			quota := d.Get("quota").(int)
+
+			if err = client.UpdateQuota(ctx, id.ShareName, quota); err != nil {
+				return fmt.Errorf("updating Quota for %s: %v", id, err)
+			}
+
+			log.Printf("[DEBUG] Updated the Quota for %s", id)
+		}
+
+		if d.HasChange("metadata") {
+			log.Printf("[DEBUG] Updating the MetaData for %s", id)
+
+			metaDataRaw := d.Get("metadata").(map[string]interface{})
+			metaData := ExpandMetaData(metaDataRaw)
+
+			if err = client.UpdateMetaData(ctx, id.ShareName, metaData); err != nil {
+				return fmt.Errorf("updating MetaData for %s: %v", id, err)
+			}
+
+			log.Printf("[DEBUG] Updated the MetaData for %s", id)
+		}
+
+		if d.HasChange("acl") {
+			log.Printf("[DEBUG] Updating the ACLs for %s", id)
+
+			aclsRaw := d.Get("acl").(*pluginsdk.Set).List()
+			acls := expandStorageShareACLsDeprecated(aclsRaw)
+
+			if err = client.UpdateACLs(ctx, id.ShareName, shares.SetAclInput{SignedIdentifiers: acls}); err != nil {
+				return fmt.Errorf("updating ACLs for %s: %v", id, err)
+			}
+
+			log.Printf("[DEBUG] Updated ACLs for %s", id)
+		}
+
+		if d.HasChange("access_tier") {
+			tier := shares.AccessTier(d.Get("access_tier").(string))
+			err = pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutUpdate), func() *pluginsdk.RetryError {
+				err = client.UpdateTier(ctx, id.ShareName, tier)
+				if err != nil {
+					if strings.Contains(err.Error(), "Cannot change access tier at this moment") {
+						return pluginsdk.RetryableError(err)
+					}
+					return pluginsdk.NonRetryableError(err)
+				}
+				time.Sleep(30 * time.Second)
+				return nil
+			})
+			if err != nil {
+				return fmt.Errorf("updating access tier %s: %+v", id, err)
+			}
+
+			log.Printf("[DEBUG] Updated Access Tier for %s", id)
+		}
+
+		return resourceStorageShareRead(d, meta)
+	}
+
+	id, err := fileshares.ParseShareID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	account, err := storageClient.FindAccount(ctx, subscriptionId, id.AccountId.AccountName)
+	existing, err := sharesClient.Get(ctx, *id, fileshares.DefaultGetOperationOptions())
 	if err != nil {
-		return fmt.Errorf("retrieving Account %q for Share %q: %v", id.AccountId.AccountName, id.ShareName, err)
+		return fmt.Errorf("retrieving %q: %v", *id, err)
 	}
-	if account == nil {
-		return fmt.Errorf("locating Storage Account %q", id.AccountId.AccountName)
+
+	model := existing.Model
+	if model == nil {
+		return fmt.Errorf("retrieving %s: model is nil", id)
 	}
 
-	// The files API does not support bearer tokens (@manicminer, 2024-02-15)
-	client, err := storageClient.FileSharesDataPlaneClient(ctx, *account, storageClient.DataPlaneOperationSupportingOnlySharedKeyAuth())
-	if err != nil {
-		return fmt.Errorf("building File Share Client for %s: %+v", account.StorageAccountId, err)
+	props := model.Properties
+	if props == nil {
+		return fmt.Errorf("retrieving %s: properties is nil", id)
 	}
```

Comment:
Same remark here regarding these checks which we typically only do if we need to patch an existing model to use for the `CreateOrUpdate` call